[Grid][Replaced] flydenver.com: Images of airlines logos have incorrect sizes.

sammygill opened the pull request:
#### 7442fff8d8744bcd2eb94face4ed371fae927c21
<pre>
[Grid][Replaced] flydenver.com: Images of airlines logos have incorrect sizes.
<a href="https://bugs.webkit.org/show_bug.cgi?id=277086">https://bugs.webkit.org/show_bug.cgi?id=277086</a>
<a href="https://rdar.apple.com/132105152">rdar://132105152</a>

Reviewed by Alan Baradlay.

A grid item&apos;s containing block is defined by its grid area. Once the
grid track sizing algorithm is done the size of this area becomes
definite and grid expresses this by setting the overriding containing
block dimensions of the grid item.

So in computeReplacedLogicalHeightUsing we should check to see if these
values are set for grid items and use them to resolve percent or
calculated logical heights.

Otherwise, we would would end up falling back to some logic that walks
up the containing block chain and could end up using the intrinsic
logical height of the image in certain scenarios which does not seem
correct for grid items.

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-in-table-cell-with-img-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-in-table-cell-with-img.html: Added.
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeReplacedLogicalHeightUsing const):

Canonical link: <a href="https://commits.webkit.org/281378@main">https://commits.webkit.org/281378@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e62f816683ae1756432362d2d9aacfc6f45de142

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59679 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39025 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12205 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63595 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10203 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61808 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46677 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10358 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48409 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7139 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61709 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36419 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51663 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29247 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33123 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8908 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9126 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55070 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9188 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65326 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3607 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9075 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/55754 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3618 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51650 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/55886 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13241 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2994 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34838 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/35921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37007 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/35666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-in-table-cell-with-img.html:
```diff
@@ -0,0 +1,27 @@
+  .cell {
+    display: table-cell;
+  }
+.grid {
```
please fix up indentation

sammygill committed:
[Grid][Replaced] flydenver.com: Images of airlines logos have incorrect sizes.
https://bugs.webkit.org/show_bug.cgi?id=277086
rdar://132105152

Reviewed by Alan Baradlay.

A grid item's containing block is defined by its grid area. Once the
grid track sizing algorithm is done the size of this area becomes
definite and grid expresses this by setting the overriding containing
block dimensions of the grid item.

So in computeReplacedLogicalHeightUsing we should check to see if these
values are set for grid items and use them to resolve percent or
calculated logical heights.

Otherwise, we would would end up falling back to some logic that walks
up the containing block chain and could end up using the intrinsic
logical height of the image in certain scenarios which does not seem
correct for grid items.

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-in-table-cell-with-img-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/grid-in-table-cell-with-img.html: Added.
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeReplacedLogicalHeightUsing const):

Canonical link: https://commits.webkit.org/281378@main
